src/context/cart/cartContext.js

```diff
@@ -0,0 +1,62 @@
+import axios from "axios";
+import React, { createContext, useContext, useReducer } from "react";
+import { cartInitialState, cartReducer } from "../../reducer/cartReducer";
+import { useAuth } from "../auth/authContext";
+
+const CartContext = createContext(cartInitialState);
+
+const getCartContextValue = () => {
+	const [cartState, cartDispatch] = useReducer(cartReducer, cartInitialState);
```

Comment:
I like your approach to this function, you can extract this into a new file also.